spec/models/user_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
```

Comment:
Great manual validations! There is a very handy ruby gem called shoulda-matchers that can be super helpful when validating attributes for a model. Definitely worth checking out some time as it's got a lot of different helpful tools built in when you have models with a lot of different attributes to validate.